data/block/miniBlockHeader.go

```diff
@@ -77,6 +77,99 @@ func (m *MiniBlockHeader) SetReserved(reserved []byte) error {
 	return nil
 }
 
+// GetProcessingType returns the miniBlock processing type as a int32
+func (m *MiniBlockHeader) GetProcessingType() int32 {
+	miniBlockHeaderReserved, err := m.getMiniBlockHeaderReserved()
+	if err != nil || miniBlockHeaderReserved == nil {
+		return int32(Normal)
+	}
+
+	return int32(miniBlockHeaderReserved.ExecutionType)
+}
+
+// SetProcessingType sets the miniBlock processing type
+func (m *MiniBlockHeader) SetProcessingType(procType int32) error {
+	var err error
+	mbhr := &MiniBlockHeaderReserved{}
+	if len(m.Reserved) > 0 {
+		mbhr, err = m.getMiniBlockHeaderReserved()
+		if err != nil {
+			return err
+		}
+	}
+	mbhr.ExecutionType = ProcessingType(procType)
+
+	return m.setMiniBlockHeaderReserved(mbhr)
+}
+
+// GetConstructionState returns the construction state of the miniBlock
+func (m *MiniBlockHeader) GetConstructionState() int32 {
+	miniBlockHeaderReserved, err := m.getMiniBlockHeaderReserved()
+	if err != nil || miniBlockHeaderReserved == nil {
+		return int32(Final)
+	}
+
+	return int32(miniBlockHeaderReserved.State)
+}
+
+// SetConstructionState sets the construction state of the miniBlock
+func (m *MiniBlockHeader) SetConstructionState(state int32) error {
+	var err error
+	mbhr := &MiniBlockHeaderReserved{}
+	if len(m.Reserved) > 0 {
+		mbhr, err = m.getMiniBlockHeaderReserved()
+		if err != nil {
+			return err
+		}
+	}
+	mbhr.State = MiniBlockState(state)
+
+	return m.setMiniBlockHeaderReserved(mbhr)
+}
+
+// IsFinal returns true if the miniBlock is final
+func (m *MiniBlockHeader) IsFinal() bool {
+	miniBlockHeaderReserved, err := m.getMiniBlockHeaderReserved()
+	if err != nil {
```

Comment:
When err != nil should be return false? Or maybe we can replace L133-L138 with:
	if err != nil || miniBlockHeaderReserved == nil {
		return true
	}
to be exactly the same as it is done in the methods above?

Reply:
right, changed the GetConstructionState but forgot to change IsFinal